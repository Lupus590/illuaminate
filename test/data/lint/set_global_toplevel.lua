-- config: (lint (only (var:set-global)) (allow-toplevel-global true))
a = 0 -- ok
function b() end -- ok
if true then c = 0 end -- ok

local function f()
  d = 0
  e = 0
end

-- We need to work out what we should do in this case. Arguably it's OK as we're
-- mutating an already defined variable. But mutating API globals within a
-- function is rather bad form.
--[[
local function f()
  a = 0
  b = 0
end
]]
