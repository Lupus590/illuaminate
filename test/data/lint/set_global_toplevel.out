set_global_toplevel.lua:[7:3-7:3]: Setting unknown global variable "d". [var:set-global]
   │
 7 │   d = 0
   │   ^
set_global_toplevel.lua:[8:3-8:3]: Setting unknown global variable "e". [var:set-global]
   │
 8 │   e = 0
   │   ^
No errors and 2 warnings
